### PR TITLE
Add release note 3.0.1

### PR DESCRIPTION
**Pull request by RobertIndie**
image
image

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: release-notes/versioned/pulsar-3.0.1.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+- [Broker] Add broker filter sync method back to guarantee the API compatibility by @Demogorgon314 in [#20826](https://github.com/apache/pulsar/pull/20826)
+- [Broker] Avoid infinite bundle unloading by @Demogorgon314 in [#20822](https://github.com/apache/pulsar/pull/20822)
+- [Broker] Call ServerCnx#closeProducer from correct thread by @michaeljmarshall in [#20747](https://github.com/apache/pulsar/pull/20747)
+- [Broker] Added the skipped message handler for ServiceUnitStateChannel by @heesung-sn in [#20677](https://github.com/apache/pulsar/pull/20677)
+- [Broker] Make ExtensibleLoadManagerImpl's broker filter pure async by @Demogorgon314 in [#20666](https://github.com/apache/pulsar/pull/20666)
+- [Broker] Fix the publish latency spike from the contention of MessageDeduplication by @codelipenghui in [#20647](https://github.com/apache/pulsar/pull/20647)
+- [Broker] Fix get owned service units NPE by @Demogorgon314 in [#20625](https://github.com/apache/pulsar/pull/20625)
+- [Broker] Topic policy can not be work well if replay policy message has any exception. by @zhanghaou in [#20613](https://github.com/apache/pulsar/pull/20613)
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:21]:
```suggestion
- [Broker] Topic policy can not work well if replay policy message has any exception. by @zhanghaou in [#20613](https://github.com/apache/pulsar/pull/20613)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+- [Broker] Fix the publish latency spike issue with large number of producers by @codelipenghui in [#20607](https://github.com/apache/pulsar/pull/20607)
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:22]:
```suggestion
- [Broker] Fix the publish latency spike issue with a large number of producers by @codelipenghui in [#20607](https://github.com/apache/pulsar/pull/20607)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+- [Broker] Do not expose bucketDelayedIndexStats by @coderzc in [#20383](https://github.com/apache/pulsar/pull/20383)
+- [Broker] Pre-create non-partitioned system topics for load balance extension by @heesung-sn in [#20370](https://github.com/apache/pulsar/pull/20370)
+- [Broker] ManagedLedger.getConfig().getProperties().putAll(properties) NPE by @heesung-sn in [#20361](https://github.com/apache/pulsar/pull/20361)
+- [Broker] Fix broker load manager class filter NPE by @Demogorgon314 in [#20350](https://github.com/apache/pulsar/pull/20350)
+- [Broker] In Key_Shared mode: remove unnecessary mechanisms of message skip to avoid unnecessary consumption stuck by @poorbarcode in [#20335](https://github.com/apache/pulsar/pull/20335)
+- [Broker] When creating a namespace anti-affinity group with the default local policies, use the bundle size supplied by the user. by @syk-coder in [#20327](https://github.com/apache/pulsar/pull/20327)
+- [Broker] Fix skip message API when hole messages exists by @crossoverJie in [#20326](https://github.com/apache/pulsar/pull/20326)
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:43]:
```suggestion
- [Broker] Fix skip message API when hole messages exist by @crossoverJie in [#20326](https://github.com/apache/pulsar/pull/20326)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+- [Broker] Gracefully shut down load balancer extension by @heesung-sn in [#20315](https://github.com/apache/pulsar/pull/20315)
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:44]:
```suggestion
- [Broker] Gracefully shut download balancer extension by @heesung-sn in [#20315](https://github.com/apache/pulsar/pull/20315)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+- [Broker] Fix class name typo `PrecisPublishLimiter` to "Precise" by @JooHyukKim in [#20310](https://github.com/apache/pulsar/pull/20310)
+- [Broker] Allow Access to System Topic Metadata for Reader Creation Post-Namespace Deletion by @liangyepianzhou in [#20304](https://github.com/apache/pulsar/pull/20304)
+- [Broker] Fix NPE cause by topic publish rate limiter. by @dragonls in [#20302](https://github.com/apache/pulsar/pull/20302)
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:47]:
```suggestion
- [Broker] Fix NPE caused by topic publish rate limiter. by @dragonls in [#20302](https://github.com/apache/pulsar/pull/20302)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+- [Broker] Fix `UnsupportedOperationException` when update topic properties. by @liangyepianzhou in [#20261](https://github.com/apache/pulsar/pull/20261)
+- [Broker] Fix ledger cachemiss size metric by @AnonHxy in [#20257](https://github.com/apache/pulsar/pull/20257)
+- [Broker] Fix default bundle size used while setting bookie affinity by @syk-coder in [#20250](https://github.com/apache/pulsar/pull/20250)
+- [Broker] Fix the behavior of delayed message in Key_Shared mode by @codelipenghui in [#20233](https://github.com/apache/pulsar/pull/20233)
+- [Broker] Fix Return value of getPartitionedStats doesn't contain subscription type by @1Jack2 in [#20210](https://github.com/apache/pulsar/pull/20210)
+- [Broker] Skip split bundle if only one broker by @poorbarcode in [#20190](https://github.com/apache/pulsar/pull/20190)
+- [Broker] Fix `RoaringBitmap.contains` can't check value 65535 by @coderzc in [#20176](https://github.com/apache/pulsar/pull/20176)
+- [Broker] Release EntryBuffer after parse proto object by @coderzc in [#20170](https://github.com/apache/pulsar/pull/20170)
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:55]:
```suggestion
- [Broker] Release EntryBuffer after parsing proto object by @coderzc in [#20170](https://github.com/apache/pulsar/pull/20170)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+### CLI
+- [Pulsar Admin] Set ns level backlog quota does not take effect if retention exists by @poorbarcode in [#20690](https://github.com/apache/pulsar/pull/20690)
+- [Pulsar Admin] Make get list from bundle Admin API async by @Demogorgon314 in [#20652](https://github.com/apache/pulsar/pull/20652)
+- [Pulsar Admin] Return BAD_REQUEST on cluster data is null for createCluster by @tisonkun in [#20346](https://github.com/apache/pulsar/pull/20346)
+- [Pulsar Admin] Fix examine messages if total message is zero by @AnonHxy in [#20152](https://github.com/apache/pulsar/pull/20152)
+- [Pulsar Admin] Fast fix infinite HTTP call getSubscriptions caused by wrong topicName by @poorbarcode in [#20131](https://github.com/apache/pulsar/pull/20131)
+- [Pulsar Admin] Report earliest msg in partitioned backlog by @teabot in [#19465](https://github.com/apache/pulsar/pull/19465)
+- [Pulsar CLI] The variable producerName of BatchMsgContainer is null by @poorbarcode in [#20819](https://github.com/apache/pulsar/pull/20819)
+- [Pulsar CLI] Allow pulser-client consume create a replicated subscription by @JooHyukKim in [#20316](https://github.com/apache/pulsar/pull/20316)
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:114]:
```suggestion
- [Pulsar CLI] Allow pulser-client consume to create a replicated subscription by @JooHyukKim in [#20316](https://github.com/apache/pulsar/pull/20316)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+- [Pulsar CLI] Fix logging noise while admin tool exit by @rdhabalia in [#19884](https://github.com/apache/pulsar/pull/19884)
+- [Client Tool] PerformanceProducer do not produce expected number of messages. by @thetumbled in [#19775](https://github.com/apache/pulsar/pull/19775)
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:116]:
```suggestion
- [Client Tool] PerformanceProducer does not produce the expected number of messages. by @thetumbled in [#19775](https://github.com/apache/pulsar/pull/19775)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+### Others
+- [Schema] Add admin cli for testCompatibility by @congbobo184 in [#19974](https://github.com/apache/pulsar/pull/19974)
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:119]:
```suggestion
- [Schema] Add admin CLI for testCompatibility by @congbobo184 in [#19974](https://github.com/apache/pulsar/pull/19974)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+- Upgrade PyYaml version to 6.0.1 by @yaalsn in [#20831](https://github.com/apache/pulsar/pull/20831)
+- Upgrade bookkeeper to 4.16.2 by @zymap in [#20704](https://github.com/apache/pulsar/pull/20704)
+- Upgrade snappy-java to address multiple CVEs by @lhotari in [#20604](https://github.com/apache/pulsar/pull/20604)
+- Upgrade Jetty to address CVEs by @lhotari in [#20461](https://github.com/apache/pulsar/pull/20461)
+- Upgrade maven surefire plugin and other build/test plugins/libs including TestNG version by @lhotari in [#20270](https://github.com/apache/pulsar/pull/20270)
+- Upgrade Netty to 4.1.93.Final by @lhotari in [#20423](https://github.com/apache/pulsar/pull/20423)
+- Upgrade sqlite-jdbc to resolve CVE-2023-32697 by @lhotari in [#20411](https://github.com/apache/pulsar/pull/20411)
+- Upgrade Guava to 32.0.0 to address CVE-2023-2976 by @lhotari in [#20459](https://github.com/apache/pulsar/pull/20459)
+
+For the complete changelog, please check: https://github.com/apache/pulsar/pulls?q=is%3Apr+label%3Arelease%2F3.0.1+
```
Comment (Anonymitaet) [inline release-notes/versioned/pulsar-3.0.1.md:131]:
```suggestion
For the complete list, check the [Pulsar 3.0.1 changelog](https://github.com/apache/pulsar/pulls?q=is%3Apr+label%3Arelease%2F3.0.1+).
```

**Comment by momo-jun** [comment]
@RobertIndie, does it make sense to also correct those PRs' titles? So that when they are cherry-picked for other releases, you won't have to go through the same issues in the coming release notes.

**Comment by RobertIndie** [comment]
> does it make sense to also correct those PRs' titles? So that when they are cherry-picked for other releases, you won't have to go through the same issues in the coming release notes.

Yes, it makes sense. The release note is to make it easy for users to check the changes. If we(maintainer) want to cherry-pick, we could use the GitHub search.

File: release-notes/versioned/pulsar-3.0.1.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+- [Broker] When creating a namespace anti-affinity group with the default local policies, use the bundle size supplied by the user. by @syk-coder in [#20327](https://github.com/apache/pulsar/pull/20327)
+- [Broker] Fix skip message API when hole messages exist by @crossoverJie in [#20326](https://github.com/apache/pulsar/pull/20326)
+- [Broker] Gracefully shut download balancer extension by @heesung-sn in [#20315](https://github.com/apache/pulsar/pull/20315)
+- [Broker] Fix class name typo `PrecisPublishLimiter` to "Precise" by @JooHyukKim in [#20310](https://github.com/apache/pulsar/pull/20310)
+- [Broker] Allow Access to System Topic Metadata for Reader Creation Post-Namespace Deletion by @liangyepianzhou in [#20304](https://github.com/apache/pulsar/pull/20304)
+- [Broker] Fix NPE caused by topic publish rate limiter. by @dragonls in [#20302](https://github.com/apache/pulsar/pull/20302)
+- [Broker] Fix `UnsupportedOperationException` when update topic properties. by @liangyepianzhou in [#20261](https://github.com/apache/pulsar/pull/20261)
```
Comment (momo-jun) [inline release-notes/versioned/pulsar-3.0.1.md:48]:
```suggestion
- [Broker] Fix `UnsupportedOperationException` when updating topic properties. by @liangyepianzhou in [#20261](https://github.com/apache/pulsar/pull/20261)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+- [Broker] Fix use error TimeUnit to record publish latency by @TakaHiR07 in [#20074](https://github.com/apache/pulsar/pull/20074)
+- [Broker] Fix tenant admin authorization bug. by @dragonls in [#20068](https://github.com/apache/pulsar/pull/20068)
+- [Broker] Fix getPartitionedStats miss subscription's messageAckRate by @wangjialing218 in [#19870](https://github.com/apache/pulsar/pull/19870)
+- [Broker] Do not filter system topic while shedding. by @thetumbled in [#18949](https://github.com/apache/pulsar/pull/18949)
+- [Broker] There are two same-named managed ledgers in the one broker by @poorbarcode in [#18688](https://github.com/apache/pulsar/pull/18688)
+- [Broker] If ledger lost, cursor mark delete position can not forward by @poorbarcode in [#18620](https://github.com/apache/pulsar/pull/18620)
+- [Broker] Support cgroup v2 by using `jdk.internal.platform.Metrics` in Pulsar Loadbalancer by @coderzc in [#16832](https://github.com/apache/pulsar/pull/16832)
+
+## Clients
+- [Client] Fix deadlock issue of consumer while using multiple IO threads by @codelipenghui in [#20669](https://github.com/apache/pulsar/pull/20669)
```
Comment (momo-jun) [inline release-notes/versioned/pulsar-3.0.1.md:76]:
```suggestion
- [Client] Fix the deadlock issue of consumes while using multiple IO threads by @codelipenghui in [#20669](https://github.com/apache/pulsar/pull/20669)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+- [Client] Make the whole grabCnx() progress atomic by @BewareMyPower in [#20595](https://github.com/apache/pulsar/pull/20595)
+- [Client] Thread-safe seek by @tisonkun in [#20321](https://github.com/apache/pulsar/pull/20321)
+- [Client] Fix  where the function getMsgNumInReceiverQueue always returns 0 when using message listener by @crossoverJie in [#20245](https://github.com/apache/pulsar/pull/20245)
+- [Client] Java Client's Seek Logic Not Threadsafe #1 by @JooHyukKim in [#20242](https://github.com/apache/pulsar/pull/20242)
+
+### Pulsar IO and Pulsar Functions
+- [Functions] Exit JVM when main thread throws exception by @michaeljmarshall in [#20689](https://github.com/apache/pulsar/pull/20689)
```
Comment (momo-jun) [inline release-notes/versioned/pulsar-3.0.1.md:83]:
```suggestion
- [Functions] Exit JVM when the main thread throws an exception by @michaeljmarshall in [#20689](https://github.com/apache/pulsar/pull/20689)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+- [Functions] Check user metric len before iterating by @flowchartsman in [#20021](https://github.com/apache/pulsar/pull/20021)
+- [Functions] Support multiple input topics for Go runtime by @jiangpengcheng in [#20000](https://github.com/apache/pulsar/pull/20000)
+- [Functions] Make pulsar-admin support update py/go with package url by @jiangpengcheng in [#19897](https://github.com/apache/pulsar/pull/19897)
+- [Functions] Fix JavaInstanceStarter inferring type class name error by @jiangpengcheng in [#19896](https://github.com/apache/pulsar/pull/19896)
+- [Functions] Fix function update error by @jiangpengcheng in [#19895](https://github.com/apache/pulsar/pull/19895)
+- [Functions] Make KubernetesRuntime translate characters in function tenant, namespace, and name during function removal to avoid label errors by @csthomas1 in [#19584](https://github.com/apache/pulsar/pull/19584)
+- [Connectors] Close the kafka source connector got stuck by @poorbarcode in [#20698](https://github.com/apache/pulsar/pull/20698)
```
Comment (momo-jun) [inline release-notes/versioned/pulsar-3.0.1.md:97]:
```suggestion
- [Connectors] Close the Kafka source connector that got stuck by @poorbarcode in [#20698](https://github.com/apache/pulsar/pull/20698)
```

File: release-notes/versioned/pulsar-3.0.1.md [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+- [Connectors] Close the kafka source connector if there is uncaught exception by @RobertIndie in [#20424](https://github.com/apache/pulsar/pull/20424)
```
Comment (momo-jun) [inline release-notes/versioned/pulsar-3.0.1.md:98]:
```suggestion
- [Connectors] Close the Kafka source connector if there is an uncaught exception by @RobertIndie in [#20424](https://github.com/apache/pulsar/pull/20424)
```

**Review by momo-jun — APPROVED** [review]
Only a couple of suggestions.

File: release-notes/versioned/pulsar-3.0.1.md (hunk 12 above)
Comment (RobertIndie) [reply release-notes/versioned/pulsar-3.0.1.md:76]: Shouldn't it be `consumers`?

Comment (momo-jun) [reply release-notes/versioned/pulsar-3.0.1.md:76]: Yes, should be `consumers`:)

**Pull request merged**